app/lib/handlers/list_intent_request_handler.rb

```diff
@@ -88,25 +88,36 @@ def prepare_item_list_by_category(category)
       items = Item.owned_by(user_id).in_category(category)
     end
 
+    def combine_duplicate_items(items)
+      items.uniq.map do |item|
+        duplicates = items.select { |i| i.name == item.name }
+        quantity = duplicates.reduce(0) {|count, i| count += i.quantity }
+        item.quantity = quantity
+      end
```

Comment:
This probably works, but it's a very slow implementation (over n^2) since you're walking over `items` once, then another time for each element in `items`, and then also walking over any of the duplicates `reduce`.

As a good exercise, try to come up with an implementation that reduces the amount of times you're iterating over things.  For bonus learning, check out Ruby's `Benchmark.measure` as a way to profile your methods.